### PR TITLE
[Merged by Bors] - chore: port order.basic again

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
There seem to have been a number of changes since this was ported, and this takes the opportunity to restore the `#align`s.

### CODE REVIEW

File: Mathlib/Order/Basic.lean [hunk 1]
```diff
@@ -979,41 +1258,49 @@ instance : LinearOrder PUnit where
 
 theorem max_eq : max a b = star :=
   rfl
+#align punit.max_eq PUnit.max_eq
 
 theorem min_eq : min a b = star :=
   rfl
+#align punit.min_eq PUnit.min_eq
 
+@[simp]
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Basic.lean:1267]:
```suggestion
-- Porting note: simp can prove this @[simp]
```

File: Mathlib/Order/Basic.lean [hunk 2]
```diff
@@ -979,41 +1258,49 @@ instance : LinearOrder PUnit where
...
 protected theorem le : a ≤ b :=
-  le_refl _
-
-theorem not_lt : ¬a < b := by
-  simp only [lt_self_iff_false]
-
+  trivial
+#align punit.le PUnit.le
 
+@[simp]
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Basic.lean:1272]:
```suggestion
-- Porting note: simp can prove this @[simp]
```

File: Mathlib/Order/Basic.lean [hunk 3]
```diff
@@ -4,17 +4,25 @@ Released under Apache 2.0 license as described in the file LICENSE.
...
+import Mathlib.Tactic.SimpRw
 import Mathlib.Tactic.Spread
 
 /-!
 # Basic definitions about `≤` and `<`
 
+> THIS FILE IS SYNCHRONIZED WITH MATHLIB4.
+> https://github.com/leanprover-community/mathlib4/pull/556
+> Any changes to this file require a corresponding PR to mathlib4.
+
```
Comment (semorrison) [inline Mathlib/Order/Basic.lean:25]:
```suggestion
```

File: Mathlib/Order/Basic.lean [hunk 4]
```diff
@@ -593,90 +737,167 @@ end OrderDual
...
+instance Pi.partialOrder [∀ i, PartialOrder (π i)] : PartialOrder (∀ i, π i) where
+  __ := Pi.preorder
+  le_antisymm := fun _ _ h1 h2 ↦ funext fun b ↦ (h1 b).antisymm (h2 b)
+#align pi.partial_order Pi.partialOrder
+
+section Pi
+
+/-- A function `a` is strongly less than a function `b`  if `a i < b i` for all `i`. -/
+def StrongLt [∀ i, LT (π i)] (a b : ∀ i, π i) : Prop :=
```
Comment (semorrison) [inline Mathlib/Order/Basic.lean:805]:
```suggestion
def StrongLT [∀ i, LT (π i)] (a b : ∀ i, π i) : Prop :=
```

**Comment by semorrison** [comment]
bors d+

**Pull request closed without merging**